File: Supragya/C/Strings/reverse_string.c
// Day56_Q2.c
// Write a c program to write a string in reverse order using recursion.
#include<stdio.h>
#include<string.h>
void reverse(char str[],int start,int end)
{
    if(start>=end)
    {
        return;
    }
    char temp=str[start];
    str[start]=str[end];
    str[end]=temp;
    reverse(str,start+1,end-1);
}
int main()
{
    char str[100];
    printf("Enter the string: ");
    scanf("%s",str);
    int len=strlen(str);
    reverse(str,0,len-1);
    printf("The reversed string is: %s",str);
    return 0;
}

File: Supragya/JAVA/Strings/reverse_string.java
// Day72_Q1.java
// Write a program to reverse a string.
import java.util.Scanner;
public class reverse_string {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Enter a string: ");
        String str = scan.nextLine();
        String rev = "";
        for(int i=0;i<str.length();i++){
            rev += str.charAt(str.length()-i-1);
        }   
        System.out.println("Reversed string: "+rev);
    }
}
